environment-api/src/main/java/com/sequenceiq/environment/api/v1/credential/model/request/CredentialRequest.java

```diff
@@ -18,6 +24,13 @@
 @JsonInclude(Include.NON_NULL)
 public class CredentialRequest extends CredentialBase {
 
+    @Size(max = 100, min = 5, message = "The length of the credential's name has to be in range of 5 to 100")
+    @Pattern(regexp = "(^[a-z][-a-z0-9]*[a-z0-9]$)",
+            message = "The name of the credential can only contain lowercase alphanumeric characters and hyphens and has start with an alphanumeric character")
+    @ApiModelProperty(value = ModelDescriptions.NAME, required = true, allowableValues = "length range[5, 100]")
+    @ResourceObjectField(action = AuthorizationResourceAction.EDIT_CREDENTIAL, variableType = AuthorizationVariableType.NAME)
```

Comment:
if we do not use this request for editing credential, only for creation, then ResourceObjectField annotation is not needed, since for creation we are using other permission to check

Reply:
thanks